src/main/groovy/liquibase/harness/compatibility/advanced/AdvancedTest.groovy

```diff
@@ -0,0 +1,188 @@
+
+            then: "check if a changelog was actually generated and validate it's content"
+            String generatedChangelog = parseQuery(readFile((String) argsMapPrimary.get("changelogFile")))
+            validateChangelog(entry.key, generatedChangelog, testInput.verificationGenCLSql, testInput.change, null)
+
+            when: "execute updateSql command on generated changelogs"
+            argsMapPrimary.put("changelogFile", resourcesDirPath + entry.value)
+            generatedSql = parseQuery(executeCommandScope("updateSql", argsMapPrimary).toString())
+            generatedSql = removeSchemaNames(generatedSql, testInput.database, testInput.primaryDbSchemaName)
+
+
+            then: "execute updateSql command on generated changelogs"
+            expectedSql = getChangelogValidationSql("generateChangelog", testInput.change, testInput.databaseName, testInput.version)
+            validateSql(generatedSql, expectedSql)
+        }
+
+        when: "execute diff command"
+        String generatedDiff = cleanDiff(executeCommandScope("diff", argsMapSecondary).toString())
+
+        then: "validate generated diff"
+        String expectedDiff = cleanDiff(getResourceContent("/$testInput.pathToExpectedDiffFile"))
+        validateDiff(generatedDiff, expectedDiff)
+
+        when: "apply generated changelog to secondary database instance and execute diff command"
+        argsMapSecondary.put("changelogFile", resourcesDirPath + generateChangelogMap.get("xmlChangelog"))
+        Scope.getCurrentScope().getUI().sendMessage("APPLY GENERATED CHANGELOG TO SECONDARY DATABASE")
+        executeCommandScope("update", argsMapSecondary)
+        generatedDiff = cleanDiff(executeCommandScope("diff", argsMapSecondary).toString())
+        expectedDiff = cleanDiff(getResourceContent("/$testInput.pathToEmptyDiffFile"))
+
+        then: "validate diff command shows no differences"
+        validateDiff(generatedDiff, expectedDiff)
+
+        when: "clear secondary instance"
+        argsMapSecondary.remove("changelogFile")
+        Scope.getCurrentScope().getUI().sendMessage("CLEAN SECONDARY DATABASE INSTANCE")
+        executeCommandScope("dropAll", argsMapSecondary)
+        argsMapSecondary.put("changelogFile", testInput.secondarySetupChangelogPath)
+
+        then: "apply specific test data to secondary instance"
+        Scope.getCurrentScope().getUI().sendMessage("SETUP SECONDARY DATABASE INSTANCE FOR DIFFCHANGELOG COMMAND")
+        executeCommandScope("update", argsMapSecondary)
+
+        for (Map.Entry<String, String> entry : diffChangelogMap.entrySet()) {
+
+            when: "execute diffChangelog command using different changelog formats"
+            argsMapSecondary.put("changelogFile", resourcesDirFullPath + entry.value)
+            executeCommandScope("diffChangelog", argsMapSecondary)
+
+            then: "check if a changelog was actually generated and validate it's content"
```

Comment:
It looks to me that it would be better to use something like 'verify diff changelog...'